src/structures/GuildChannel.js

```diff
@@ -366,12 +358,12 @@ class GuildChannel extends Channel {
 
     const newData = await this.client.api.channels(this.id).patch({
       data: {
-        name: (data.name || this.name).trim(),
-        type: ChannelTypes[data.type?.toUpperCase()],
+        name: (data.name ?? this.name).trim(),
+        type: ChannelTypes[(data.type ?? this.type).toUpperCase()],
```

Comment:
Reverterino this line.
```suggestion
        type: ChannelTypes[data.type?.toUpperCase()],
```